player.gd

```diff
@@ -57,31 +57,36 @@ func _force_update_is_on_floor():
 	velocity = old_velocity
 
 func _on_rolling_timer_timeout():
-	roll = false
+	is_rolling = false
 
-func _create_rolling_timer(duration):
-	var timer = get_tree().create_timer(duration)
-	#timer.connect("timeout", timer, "_on_rolling_timer_timeout")
-	#timer.start()
 
 @rpc("any_peer", "call_local", "unreliable")
 func play_animation():
-	var _is_flipped = input.mouse_coordinates[0] < _animated_sprite.global_position.x
-	_animated_sprite.set_flip_h(_is_flipped)
+	var _is_facing_left = input.mouse_coordinates[0] < _animated_sprite.global_position.x
+	var _is_moving_left = input.direction.x < 0
+	_animated_sprite.set_flip_h(_is_facing_left)
 	if movement_state == MovementState.IDLE:
 		_animated_sprite.play("IDLE")
 	elif movement_state == MovementState.JUMPING:
 		_animated_sprite.play("JUMPING")
 	elif movement_state == MovementState.RUNNING:
-		_animated_sprite.play("RUNNING")
+		if _is_facing_left != _is_moving_left:
+			_animated_sprite.play_backwards("RUNNING")
+		else:
+			_animated_sprite.play("RUNNING")
 	elif movement_state == MovementState.CROUCH_WALK:
-		_animated_sprite.play("CROUCH_WALK")
+		if _is_facing_left != _is_moving_left:
+			_animated_sprite.play_backwards("CROUCH_WALK")
+		else:
+			_animated_sprite.play("CROUCH_WALK")
 	elif movement_state == MovementState.CROUCH_IDLE:
 		_animated_sprite.play("CROUCH_IDLE")
 	elif movement_state == MovementState.ROLLING:
 		_animated_sprite.play("ROLLING")
 	elif movement_state == MovementState.CLIMBING:
 		_animated_sprite.play("CLIMBING")
+		if input.direction.y == 0:
+			_animated_sprite.pause()
```

Comment:
We might want a different state to handle this, cuz in bman we can shoot certain weapons while hanging on a ladder which has a diffrent animation with one arm on the ladder and weapon in the other

Reply:
I think we should use states for physics states not animation states. When we add shooting from ladder I think we can check which weapon is equipped and add logic to change that. Open to suggestions though